docs/tutorial.rst

```diff
@@ -0,0 +1,172 @@
+.. role:: math(raw)
+   :format: html latex
+..
+
+.. role:: raw-latex(raw)
+   :format: latex
+..
+
+FPYLLL
+======
+
+::
+
+  >>> from fpylll import IntegerMatrix
+  >>> A = IntegerMatrix(9,10)
+  >>> A.randomize("intrel", bits=10)
+
+Matrix :math:`A` is a (random) knapsack type matrix. That is of the form  `[ {\bf a} | I_n]`, where `{\bf a}` is a column vector of dimension `n`, and `I_n` the `n`-dimensional identity matrix. Giving
+
+::
+
+  >>> print A
+  [ 116 1 0 0 0 0 0 0 0 0 ]
+  [ 331 0 1 0 0 0 0 0 0 0 ]
+  [ 303 0 0 1 0 0 0 0 0 0 ]
+  [ 963 0 0 0 1 0 0 0 0 0 ]
+  [ 456 0 0 0 0 1 0 0 0 0 ]
+  [ 225 0 0 0 0 0 1 0 0 0 ]
+  [ 827 0 0 0 0 0 0 1 0 0 ]
+  [ 381 0 0 0 0 0 0 0 1 0 ]
+  [  99 0 0 0 0 0 0 0 0 1 ]
+
+    
+Also, the following type of matrices are supported,
+
+::
+
+  >>> b = 10
+  >>> p = 521 # prime
+  >>> A = IntegerMatrix(10,10)
+  >>> A.randomize("uniform", bits=b)
+  >>> A.randomize("ntrulike", bits=b, q=p)
+  >>> A.randomize("ntrulike2", bits=b, q=p)
+
+For a user defined matrix we use the method **from\_matrix**
+
+::
+
+  >>> A = IntegerMatrix.from_matrix([[1,2,3,4],[30,4,4,5],[1,-2,3,4],[0,0,1,0]])
+
+
+  To compute the Gram-Schimdt form of the matrix `{\bf A}`, we use the **GSO** class
+
+::
+
+  >>> from fpylll import GSO
+  >>> M = GSO.Mat(A)
+  >>> # to write a vector v as a linear combination of the GS-basis of RowSp(A):
+  >>> v = (1,2,5,5)
+  >>> _ = M.update_gso()
+  >>> print(M.from_canonical(v))
+  (1.3333333333333333, -0.013019739605207909, 0.1949374454466105, 1.2521739130434784)
+  >>> # the dimension of the GS-matrix :
+  >>> print(M.d)
+  4
+
+To compute the inner product `{\bf b}_i\cdot {\bf b}^{*}_j` and the coefficient `\mu_{i,j} = {\bf b}_i\cdot {\bf b}^*_j / ||{\bf b}^*_j||^2`
+
+::
+
+  >>> i = 3; j = 2;
+  >>> print M.get_r(i,j)
+  0.810079798404
+  >>> print M.get_mu(i,j)
+  0.0584569876831
+
+  To compute the LLL reduced matrix of `{\bf A}`
+
+::
+
+  >>> from fpylll import LLL
+  >>> A_lll = LLL.reduction(A)
+  >>> print A_lll
+  [  0  0 1  0 ]
+  [  0 -4 0  0 ]
+  [  1  2 0  4 ]
+  [ 27 -2 0 -7 ]
+
+
+For the BKZ reduction of `{\bf A}` with blocksize say 5 (without pruning),
+
+::
+
+  >>> from fpylll import BKZ
+  >>> block_size = 5
+  >>> A_bkz = BKZ.reduction(A, BKZ.Param(block_size) )
```

Comment:
Remove space at end. Our tutorials etc. will inform user's style, especially if they're just starting with Python. Might as well get them used to PEP8